### PR TITLE
impl core::fmt::Debug for hmac::SigningKey and hmac::SigningContext

### PR DESCRIPTION
**Pull request by andrewtj**
I don't have any particular preference for the output so I based it on #541.

### CODE REVIEW

**Comment by briansmith** [comment]
Does `derive_debug_from_field!` work for this use? If so, let's use it. Otherwise, why not?

**Comment by andrewtj** [comment]
Another option would be to modify `derive_debug_from_field!` to take `$($tt:tt)*` instead of an ident - would that be better?

**Comment by briansmith** [comment]
> Another option would be to modify `derive_debug_from_field!` to take `$($tt:tt)*` instead of an ident - would that be better?

That does sound better to me in the abstract, assume we rename the `from_field` part.

**Comment by andrewtj** [comment]
> That does sound better to me in the abstract, assume we rename the `from_field` part.

Okay. I've pushed 7fd1fa7 which renames the macro to `derive_debug_via_self!` and also makes the macro take `self.something` as the second param. If either change isn't too taste I'm happy to make further changes.

File: src/debug.rs [hunk 1]
```diff
@@ -14,11 +14,11 @@
 
 // Generates an implementation of the Debug trait for a type that defers to the
 // Debug implementation for a given field.
-macro_rules! derive_debug_from_field {
-    ($typename:ty, $fieldname:ident) => {
+macro_rules! derive_debug_via_self {
+    ($typename:ty, self.$($tt:tt)+) => {
```
Comment (briansmith) [inline src/debug.rs]: Interesting, I didn't realize we could enforce that the expression starts with `self` like this.

**Pull request merged**